package/kedro_viz/api/graphql.py

```diff
@@ -208,57 +209,113 @@ def format_run_tracking_data(
     return JSONObject(formatted_tracking_data)
 
 
+def get_tracking_datasets() -> list:
+
+    """Gets all the versioned datasets from the catalog that need to be tracked in experiments; i.e.
+    plot datasets such as Plotly, Matplotlib, and tracking datasets such as Metrics, JSON Datasets
+
+    Returns:
+        List of all versioned datasets
+    """
+
+    # pylint: disable=import-outside-toplevel
+    from kedro.extras.datasets.matplotlib import MatplotlibWriter
+    from kedro.extras.datasets.plotly import JSONDataSet as PlotlyJSONDataSet
+    from kedro.extras.datasets.plotly import PlotlyDataSet
+    from kedro.extras.datasets.tracking import JSONDataSet, MetricsDataSet  # noqa: F811
+
+    tracking_datasets = []
+
+    catalog = data_access_manager.catalog.get_catalog()
+    for ds_name, ds_value in catalog._data_sets.items():
+        ds_type = ""
+        if isinstance(ds_value, MetricsDataSet):
+            ds_type = "Metrics"
+        elif isinstance(ds_value, JSONDataSet):
+            ds_type = "JSON Data"
+        elif isinstance(ds_value, (PlotlyJSONDataSet, PlotlyDataSet)):
+            ds_type = "Plotly"
+        elif isinstance(ds_value, (MatplotlibWriter)):
+            ds_type = "Matplotlib"
+        else:
+            continue  # pragma: no cover
```

Comment:
yes, we don't but we do check later on it, if that particular dataset belongs to a run and if it's not a versioned dataset it doesn't get added because it has no run_id associated with it